python/lsst/pipe/tasks/compute_object_epochs.py

```diff
@@ -0,0 +1,133 @@
+# This file is part of pipe_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from astropy.table import Table
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import numpy as np
+
+
+class ComputeObjectEpochsConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=("tract", "skymap"),
```

Comment:
I wonder if this should be `patch`, instead, given that it takes one patch-level input catalog and produced one patch-level output catalog, and the tract-level `epochMap` can't be that expensive to load once per patch.

Reply:
I'm not actually sure what the connections-level dimensions do so I defer to your expertise on that.

Reply:
It sets the dimensions of a single quantum, so the level of parallelization the task runs at.  I do think `patch` makes more sense, but that will require changes to other parts of the task; the input and output catalogs would be `multiple=False` and hence they wouldn't appear to `runQuantum` as lists anymore.

Reply:
Oh, I see. I can't see any other reason it would've been written this way other than to avoid loading the epoch map for every patch. It seems almost certain to be I/O bound, though, so is there some other benefit to having deferred loading of a known set of patch tables vs running them all independently?

Reply:
Running them separately makes error- and missing-data handling a lot easier, since you don't have to worry about making sure one bad patch doesn't bring down the others.  It's also just a lot more familiar; usually when we have a task that does a gather-scatter like this, it's because there algorithmically needs to be a gather-scatter, and here that's just not true.

Reply:
Alright. Is there any point in keeping either the `deferLoad` or `deferGraphConstraint` options, then?

Reply:
I think you still want `deferLoad=True` to make it easier to load just a few columns.  I don't think you need `deferGraphConstraint`, but I'm not actually sure why that was there before, either.